smt/ctx.cpp

```diff
@@ -9,7 +9,7 @@ namespace smt {
 
 context ctx;
 
-void context::init() {
+void context::initialize() {
```

Comment:
This is problematic. Z3 also has an smt::context class, so this can be a mess in the long run.
FOr dynamic linking is fine, because symbols are hidden with the default build options. Why are statically linking Z3? Is this a useful use case?  (If so, this fix is not a future-proof one)

Reply:
there's a weird cmake problem on OSX where, no matter what we try, it is pulling the system-installed libz3.dylib no matter what we do. sometimes this is OK but other times it creates problems, and I'd like to avoid it.

since nothing we've tried fixes the cmake thing, linking z3 statically seems like an OK workaround to make sure we get the one that we want.

Reply:
in other words, all things like this are not working:
```
-DZ3_LIBRARIES=/Users/johnregehr/souper-regehr/third_party/z3-install/lib/libz3.dylib
```